File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-12-01/examples-js/Common/BackupResourceVaultConfigs_Put.js
const { RecoveryServicesBackupClient } = require("@azure/arm-recoveryservicesbackup");
const { DefaultAzureCredential } = require("@azure/identity");

async function updateVaultSecurityConfig() {
  const subscriptionId = "00000000-0000-0000-0000-000000000000";
  const vaultName = "SwaggerTest";
  const resourceGroupName = "SwaggerTestRg";
  const parameters = {
    properties: {
      enhancedSecurityState: "Enabled",
      softDeleteFeatureState: "Enabled",
    },
  };
  const credential = new DefaultAzureCredential();
  const client = new RecoveryServicesBackupClient(credential, subscriptionId);
  const result = await client.backupResourceVaultConfigs.put(
    vaultName,
    resourceGroupName,
    parameters
  );
  console.log(result);
}

updateVaultSecurityConfig().catch(console.error);

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2025-02-01/examples-java/Common/BackupResourceVaultConfigs_Put.java

import com.azure.resourcemanager.recoveryservicesbackup.fluent.models.BackupResourceVaultConfigResourceInner;
import com.azure.resourcemanager.recoveryservicesbackup.models.BackupResourceVaultConfig;
import com.azure.resourcemanager.recoveryservicesbackup.models.EnhancedSecurityState;
import com.azure.resourcemanager.recoveryservicesbackup.models.SoftDeleteFeatureState;

/**
 * Samples for BackupResourceVaultConfigs Put.
 */
public final class Main {
    /*
     * x-ms-original-file:
     * specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2025-02-01/examples/
     * Common/BackupResourceVaultConfigs_Put.json
     */
    /**
     * Sample code: Update Vault Security Config.
     * 
     * @param manager Entry point to RecoveryServicesBackupManager.
     */
    public static void updateVaultSecurityConfig(
        com.azure.resourcemanager.recoveryservicesbackup.RecoveryServicesBackupManager manager) {
        manager.backupResourceVaultConfigs().putWithResponse("SwaggerTest", "SwaggerTestRg",
            new BackupResourceVaultConfigResourceInner()
                .withProperties(new BackupResourceVaultConfig().withEnhancedSecurityState(EnhancedSecurityState.ENABLED)
                    .withSoftDeleteFeatureState(SoftDeleteFeatureState.ENABLED)),
            com.azure.core.util.Context.NONE);
    }
}
